app/controllers/amp_articles_controller.rb

```diff
@@ -0,0 +1,47 @@
+class AmpArticlesController < ActionController::Base
+  def show
+    if (@article = Article.find_by_permalink(params[:from])).present?
+      if @article.supports_amp || params[:no_redirect]
+        respond_to do |format|
+          format.html do
+            render 'articles/amp/show'
+          end
+        end
+      else
+        redirect_to article_url(@article.permalink)
+      end
+    else
+      render 'errors/404', status: 404
+    end
+  end
+
+  def article_body(article)
+    body = article.html(:body)
+    page = Nokogiri::HTML.fragment(body)
+    page = process_img_tag(page)
+    page = process_iframe_tag(page)
+    page
+  end
+  helper_method :article_body
```

Comment:
Can we move this to a decorator rather than a helper method? Or move to a Plain ruby object that does this and the private methods above? IMHO, this is not responsibility of the controller process the img neither the iframe tag, this could be something AMPProcessor responsibility or something like that.